object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionTransfer.java

```diff
@@ -167,9 +227,41 @@ public void accumulate(Node node, TransferResult<CFValue, CFStore> result, Strin
           }
         }
       }
+      AnnotationMirror newAnno = atypefactory.createAccumulatorAnnotation(valuesAsList);
+      exceptionalStores.values().stream().forEach(s -> s.insertValue(target, newAnno));
     }
-    AnnotationMirror newAnno = atypefactory.createAccumulatorAnnotation(valuesAsList);
-    exceptionalStores.values().stream().forEach(s -> s.insertValue(target, newAnno));
+  }
+
+  private void insertIntoStores(
+      TransferResult<CFValue, CFStore> result, JavaExpression target, AnnotationMirror newAnno) {
+    if (result.containsTwoStores()) {
+      CFStore thenStore = result.getThenStore();
+      CFStore elseStore = result.getElseStore();
+      thenStore.insertValue(target, newAnno);
+      elseStore.insertValue(target, newAnno);
+    } else {
+      CFStore store = result.getRegularStore();
+      store.insertValue(target, newAnno);
+    }
+  }
+
+  protected VariableTree createTemporaryVar(Node method) {
+    TypeMirror iteratorType = TreeUtils.typeOf(method.getTree());
+    Element ee = TreeUtils.elementFromTree(method.getTree());
+    // Declare and initialize a new, unique iterator variable
+    VariableTree iteratorVariable =
+        treeBuilder.buildVariableDecl(
+            iteratorType, // annotatedIteratorTypeTree,
+            uniqueName("tmp"),
+            ee.getEnclosingElement(),
+            (ExpressionTree) method.getTree());
+    return iteratorVariable;
+  }
+
+  protected long uid = 0;
+
+  protected String uniqueName(String prefix) {
+    return prefix + "#num" + uid++;
```

Comment:
This code seems to be copy-pasted based on local variable names.  It may be hard to share the logic, but at least rename the variables to be appropriate to this usage.

Reply:
I renamed the variables to "temp-var-id"